src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php

```diff
@@ -376,7 +378,7 @@ public function getTimeZoneId()
         }
 
         // In PHP 5.5 default timezone depends on `date_default_timezone_get()` method
-        if (PHP_VERSION_ID >= 50500) {
+        if (PHP_VERSION_ID >= 50500 || (extension_loaded('intl') && method_exists('IntlDateFormatter', 'setTimeZone'))) {
```

Comment:
without this, the fallback implementation can't be compared to the currently loaded intl when one is loaded